store/sqlstore/post_store.go

```diff
@@ -495,19 +515,49 @@ func (s *SqlPostStore) Delete(postId string, time int64, deleteByID string) erro
 		return errors.Wrap(err, "failed to update Posts")
 	}
 
-	return nil
+	return s.cleanupThreads(post.Id, post.RootId, post.UserId)
```

Comment:
Since cleaning up child thread posts is not an invariant in the system (for example in the case of data retention where you delete the parent but not the children), coupled with the fact that this performs another separate SQL query, I think this should be performed in the App tier rather than here in the data tier.

Reply:
Also, was this a breaking change to the delete post API endpoint or did it always also delete child posts?

Reply:
@mkraft - this was not a breaking change. Regarding data retention - we're aware that collapsed threads do not work with retention enabled at the moment, we have a ticket tracking it